### PR TITLE
[1LP][RFR] Expect 200 or 204 from a REST DELETE, change to using assert_response() in most places

### PR DESCRIPTION
**Pull request by bsquizz**
These tests were failing due to MIQ API returning a 200 OK after calling delete. Technically, this is acceptable according to https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html section 9.7

### CODE REVIEW

File: cfme/tests/cloud/test_providers.py [hunk 1]
```diff
@@ -478,10 +478,9 @@ def test_delete_arbitration_profiles_from_detail(self, appliance, arbitration_pr
         Metadata:
             test_flag: rest
         """
-        status = 204 if method == 'delete' else 200
         for entity in arbitration_profiles:
             entity.action.delete(force_method=method)
-            assert appliance.rest_api.response.status_code == status
+            assert appliance.rest_api.response.status_code in [200, 204]
```
Comment (RonnyPfannschmidt) [inline cfme/tests/cloud/test_providers.py]: **optional** please use a tuple expression

**Comment by mfalesni** [comment]
@RonnyPfannschmidt Some time ago I was experimenting with such pattern which could be more useful in testing https://gist.github.com/mfalesni/9dba8c00856ee093fc5aea84ae7f3161#file-rest_handler-py-L136

**Comment by psav** [comment]
Yes it was something I was going to discuss with @mkoura. It's good, but perhaps not right now, there are other priorities which take precedence

File: cfme/tests/cloud/test_providers.py (hunk 1 above)
Comment (mkoura) [reply cfme/tests/cloud/test_providers.py]: REQUIRED: please use ``assert_response(appliance)`` everywhere. This is legacy code that is being replaced by ``assert_response`` that correctly checks what status is expected.

**Comment by mkoura** [comment]
@RonnyPfannschmidt I have plans to do this for deletes using single resource as well as for edits (both using single resource and collection). I already replaced deletes from collection with https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/rest.py#L113 in all tests.

**Comment by bsquizz** [comment]
@mkoura I've updated assert_response to accept an int or tuple/list of ints for the `http_status` kwarg. I also wrote a docstring for that method to describe its behavior, and anywhere else I found us using the old style status code assertions I replaced it with `assert_response(appliance, http_status=<status>)` because that seemed like something easy to take care of ...

Waiting for PRT to run on it.

**Comment by mkoura** [comment]
@bsquizz I don't think this is necessary. There's well defined behavior that we can check - if there's a body in the response, success status code is 200. If there's no body, success status code is 204. The ``assert_response`` checks that.
The response can change, as you can see by these failing tests - the request used to return 204 as there was no body in the response, now they are returning 200 because there is a body. I don't think hardcoding expected status is good idea. We don't need it for checking correct behavior.

**Comment by bsquizz** [comment]
Updated to simply `assert_response(appliance)` where previously we were explicitly expecting status code == 200, 204, etc.

**Comment by bsquizz** [comment]
OK I see now, it was news to me that bool() on a Requests.Response instance returns False if it's a non-success code...

File: cfme/tests/configure/test_tag_category.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def test_create_categories(self, appliance, categories):
         """
         for ctg in categories:
             record = appliance.rest_api.collections.categories.get(id=ctg.id)
-            assert appliance.rest_api.response.status_code == 200
-            assert record.name == ctg.name
+        assert_response(appliance)
+        assert record.name == ctg.name
```
Comment (mkoura) [inline cfme/tests/configure/test_tag_category.py]: REQUIRED: indentation is wrong, this should be inside of the loop

**Comment by dajoRH** [comment]
Lint report for commit 4aadd777b3bc159b00cfc626968bff2248ceaf86:

`cfme/utils/rest.py`:
- :heavy_exclamation_mark: Line 15:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by bsquizz** [comment]
@mkoura how's this looking to you now?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/configure/test_tag_category.py (hunk 2 above)
Comment (bsquizz) [reply cfme/tests/configure/test_tag_category.py]: fixed

**Comment by dajoRH** [comment]
Lint report for commit f8dbc54a5eb47dd34f9b71826f80f9c63f7c9b81:

`cfme/tests/cloud/test_providers.py`:
- :red_circle: Line 417:9: **F821** *undefined name 'assert_response'*
- :red_circle: Line 435:9: **F821** *undefined name 'assert_response'*
- :red_circle: Line 469:13: **F821** *undefined name 'assert_response'*
- :red_circle: Line 485:13: **F821** *undefined name 'assert_response'*
- :red_circle: Line 488:13: **F821** *undefined name 'assert_response'*
- :red_circle: Line 501:9: **F821** *undefined name 'assert_response'*
- :red_circle: Line 504:9: **F821** *undefined name 'assert_response'*
- :red_circle: Line 524:17: **F821** *undefined name 'assert_response'*
- :red_circle: Line 529:13: **F821** *undefined name 'assert_response'*
- :red_circle: Line 558:9: **F821** *undefined name 'assert_response'*
- :red_circle: Line 584:9: **F821** *undefined name 'assert_response'*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by bsquizz** [comment]
Just going to confirm the PRT run was clean

**Comment by bsquizz** [comment]
Checked PRT report, failures aren't related to these changes

**Pull request merged**